packages/ember-metal/lib/streams/stream.js

```diff
@@ -85,43 +110,51 @@ Stream.prototype = {
   },
 
   subscribe: function(callback, context) {
-    if (this.subscribers === undefined) {
-      this.subscribers = [callback, context];
+    var subscriber = new Subscriber(callback, context, this);
+    if (this.subscriberHead === null) {
+      this.subscriberHead = this.subscriberTail = subscriber;
     } else {
-      this.subscribers.push(callback, context);
+      var tail = this.subscriberTail;
+      tail.next = subscriber;
+      subscriber.prev = tail;
+      this.subscriberTail = subscriber;
     }
+
+    var stream = this;
+    return function() { subscriber.removeFrom(stream); };
```

Comment:
Another note on this one: it might be good to save the parent subscriber and put the remove function on the prototype. Otherwise we'll be allocating a ton of functions/closures in what seems to be a hot path for views.